pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -195,6 +195,51 @@ func (pl *VolumeBinding) isSchedulableAfterPersistentVolumeClaimChange(logger kl
 	return framework.QueueSkip, nil
 }
 
+// isSchedulableAfterCSIStorageCapacityChange checks whether a CSIStorageCapacity event might make a Pod schedulable or not.
+// Any CSIStorageCapacity addition and a CSIStorageCapacity update to capacity and maximumVolumeSize
+// might make a Pod schedulable.
+// Note that an update to nodeTopology and storageClassName is not allowed and we don't have to consider while examining the update event.
+func (pl *VolumeBinding) isSchedulableAfterCSIStorageCapacityChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldCap, newCap, err := util.As[*storagev1.CSIStorageCapacity](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"CSIStorageCapacity", klog.KObj(newCap),
+	)
+
+	if oldCap == nil {
+		logger.V(5).Info("A new CSIStorageCapacity was created, which could make a Pod schedulable")
+		return framework.Queue, nil
+	}
+
```

Comment:
Please revisit https://github.com/kubernetes/kubernetes/blob/1904a56e3071035de19016cef621866a37140d5f/pkg/scheduler/framework/plugins/volumebinding/binder.go#L1050-L1057, I believe we can do further.

Probably it's better to do like -

```go
func (pl *VolumeBinding) isSchedulableAfterCSIStorageCapacityChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
//...

	oldLimit := volumeLimit(oldCap)
	newLimit := volumeLimit(newCap)
	
	if oldLimit != nil && newLimit == nil { return QueueSkip }
	if oldLimit == nil && newLimit != nil { return Queue }
	if oldLimit.Value() < newLimit.Value() { return Queue }
	return QueueSkip
}

func volumLimit(capacity) *resource.Quantity {
	if capacity.MaximumVolumeSize != nil {
		// Prefer MaximumVolumeSize if available, it is more precise.
		return capacity.MaximumVolumeSize
	}
	return capacity.Capacity
}
```

Reply:
@sanposhiho I've made the corrections. Please review again.